src/ucp/core/ucp_mm.c

```diff
@@ -283,18 +283,22 @@ static ucs_status_t ucp_mem_map_common(ucp_context_h context, void *address,
     memh->length  = length;
 
     if (is_allocate) {
+        if (!((memory_type == UCS_MEMORY_TYPE_HOST) ||
+              (memory_type == UCS_MEMORY_TYPE_UNKNOWN))) {
```

Comment:
why not just `if (memory_type != UCS_MEMORY_TYPE_HOST)`?
for memory allocation, ucp cannot detect the type of non-existent memory) (or need comment in API doc that for memory allocation type, if not specified, is HOST)?

Reply:
in the future, we may add allocation with specified memory type. 
is it ok with text in API doc saying "If this value is not set with memory allocate request, it allocates host memory @ ref UCS_MEMORY_TYPE_HOST"

Reply:
so, UCS_MEMORY_TYPE_UNKNOWN is also invalid value for allocation.
```
For memory allocation: If this value is not set (along with its corresponding bit in the field_mask -
@ref UCP_MEM_MAP_PARAM_FIELD_FLAGS), @ref UCS_MEMORY_TYPE_HOST will be used.
For memory registration: If this value is not set (along with its corresponding bit in the field_mask -
@ref UCP_MEM_MAP_PARAM_FIELD_FLAGS) or set to @ref UCS_MEMORY_TYPE_UNKNOWN,
ucp will internally detect the memory type.
```